packages/fuchsia_remote_debug_protocol/test/src/runners/ssh_command_runner_test.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io' show ProcessResult;
+
+import 'package:test/test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:process/process.dart';
+import 'package:fuchsia_remote_debug_protocol/src/runners/ssh_command_runner.dart';
+
+void main() {
+  group('SshCommandRunner.constructors', () {
+    test('throws exception with invalid address', () async {
+      SshCommandRunner newCommandRunner() {
+        return new SshCommandRunner(address: 'sillyaddress.what');
+      }
+
+      expect(newCommandRunner, throwsArgumentError);
+    });
+
+    test('throws exception from injection constructor with invalid addr',
+        () async {
+      SshCommandRunner newCommandRunner() {
+        return new SshCommandRunner.withProcessManager(
+            const LocalProcessManager(),
+            address: '192.168.1.1.1');
+      }
+
+      expect(newCommandRunner, throwsArgumentError);
+    });
+  });
+
+  group('SshCommandRunner.run', () {
+    MockProcessManager mockProcessManager;
+    MockProcessResult mockProcessResult;
+    SshCommandRunner runner;
+
+    setUp(() {
+      mockProcessManager = new MockProcessManager();
+      mockProcessResult = new MockProcessResult();
+      when(mockProcessManager.run(typed(any))).thenReturn(mockProcessResult);
+    });
+
+    test('verify interface is appended to ipv6 address', () async {
+      final String ipV6Addr = 'fe80::8eae:4cff:fef4:9247';
+      final String interface = 'eno1';
+      runner = new SshCommandRunner.withProcessManager(
+        mockProcessManager,
+        address: ipV6Addr,
+        interface: interface,
+        sshConfigPath: '/whatever',
+      );
+      when<String>(mockProcessResult.stdout).thenReturn('somestuff');
+      when(mockProcessResult.exitCode).thenReturn(0);
+      await runner.run('ls /whatever');
+      final List<String> passedCommand =
+          verify(mockProcessManager.run(captureAny)).captured.single;
+      expect(passedCommand.indexOf('$ipV6Addr%$interface'), isNot(-1));
```

Comment:
See also https://www.dartdocs.org/documentation/test/0.12.31/test/contains.html (this comment applies throughout the file)

Reply:
Done.